src/zkevm_specs/evm/execution/create.py

```diff
@@ -0,0 +1,149 @@
+from zkevm_specs.evm.opcode import Opcode
+from zkevm_specs.util.arithmetic import FQ, RLC
+from zkevm_specs.util.hash import EMPTY_CODE_HASH, EMPTY_HASH
+from zkevm_specs.util.param import (
+    GAS_COST_CREATE,
+    N_BYTES_GAS,
+    N_BYTES_U64,
+)
+from ...util import (
+    CALL_CREATE_DEPTH,
+)
+from ..instruction import Instruction, Transition
+from ..table import RW, CallContextFieldTag, AccountFieldTag, CopyDataTypeTag
+
+
+def create(instruction: Instruction):
+    opcode = instruction.opcode_lookup(True)
+    is_create, is_create2 = instruction.pair_select(opcode, Opcode.CREATE, Opcode.CREATE2)
+    instruction.responsible_opcode_lookup(opcode)
+    callee_call_id = instruction.curr.rw_counter
+
+    value = instruction.stack_pop()
+    offset = instruction.stack_pop()
+    size = instruction.stack_pop()
+    contract_address = instruction.stack_push()
+
+    depth = instruction.call_context_lookup(CallContextFieldTag.Depth)
+    tx_id = instruction.call_context_lookup(CallContextFieldTag.TxId)
+    caller_address = instruction.call_context_lookup(CallContextFieldTag.CallerAddress)
+    nonce, nonce_prev = instruction.account_write(caller_address, AccountFieldTag.Nonce)
+    is_success = instruction.call_context_lookup(CallContextFieldTag.IsSuccess)
+    is_static = instruction.call_context_lookup(CallContextFieldTag.IsStatic)
+    reversion_info = instruction.reversion_info()
+    contract_address = instruction.generate_contract_address(caller_address, nonce_prev)
+
+    # Verify depth is less than 1024
+    instruction.range_lookup(depth, CALL_CREATE_DEPTH)
+
+    # ErrNonceUintOverflow constraint
+    (is_not_overflow, _) = instruction.compare(nonce, nonce_prev, N_BYTES_U64)
+    instruction.is_zero(is_not_overflow)
+
+    # add contract address to access list
+    instruction.add_account_to_access_list(tx_id, contract_address)
+
+    # ErrContractAddressCollision constraint
+    code_hash = instruction.account_read(contract_address, AccountFieldTag.CodeHash)
+    instruction.constrain_equal(code_hash, RLC(EMPTY_HASH))
```

Comment:
It seems we need to update the account's `code_hash` from `0` to `RLC(EMPTY_HASH)`, since the account to be deployed should be non-existing and we need to create it.

Reply:
I guess the account's `code_hash` would be either 0 or `EMPTY_HASH` since anyone could transfer ETH to the address before the contract deployment, right? So, the contraint here should be either 0 or `RLC(EMPTY_HASH)`?

Reply:
ah that's true, you are right!

Reply:
fixed in [98f9722](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/355/commits/98f972291b6884c84227ca761e668e5999c2c525)